internal/stores/metadata.go

```diff
@@ -248,6 +271,7 @@ func (s *SQLStore) AddContract(ctx context.Context, c rhpv2.ContractRevision, to
 	if err != nil {
 		return api.ContractMetadata{}, err
 	}
+	s.isOurContract[c.ID()] = struct{}{}
```

Comment:
good candidate for gorm hooks - this is begging to be forgotten by someone that inserts/adds contracts

Reply:
gorm hooks don't have access to the map